### PR TITLE
Add log-level command line switch

### PR DESCRIPTION
**Pull request by manno**
* `--log-level=` controls minimal log level
* prefix controller-runtime logs with 'cr',
  that seems to result in 'cr.kubebuilder' prefixed log entries.
* configure caller skip for predicate related logging, so we see
  the actual caller location in the logs

[#167086778](https://www.pivotaltracker.com/story/show/167086778)

### CODE REVIEW

File: pkg/kube/util/ctxlog/context.go [hunk 1]
```diff
@@ -34,6 +34,11 @@ func NewContextWithRecorder(ctx context.Context, name string, recorder record.Ev
 	return context.WithValue(ctx, ctxLoggerKey, log)
 }
 
+func ExtractLoggerWithSkip(ctx context.Context, skip int) *zap.SugaredLogger {
```
Comment (edwardstudy) [inline pkg/kube/util/ctxlog/context.go]: We should give a comment if we follow go lint.

File: cmd/internal/root.go [hunk 2]
```diff
@@ -132,14 +137,20 @@ func init() {
 
 // newLogger returns a new zap logger
 func newLogger(options ...zap.Option) *zap.SugaredLogger {
-
-	logger, err := zap.NewDevelopment(options...)
+	level := viper.GetString("log-level")
+	l := zap.DebugLevel
+	l.Set(level)
+
+	cfg := zap.NewDevelopmentConfig()
```
Comment (edwardstudy) [inline cmd/internal/root.go]: Why we don't use `NewProductionConfig` directly if we don't want to keep development env? Do we have customized something?

Comment (manno) [reply cmd/internal/root.go]: It's very different from the log format we have now: https://github.com/uber-go/zap/blob/master/config.go#L114-L127
Judging from the code it outputs everything as JSON.
I tried it a few months back and if I remember correctly, it doesn't contain code locations.

Setting `Development: false` just makes the stack traces more dense, according to documentation.

File: pkg/kube/util/ctxlog/context.go (hunk 1 above)
Comment (manno) [reply pkg/kube/util/ctxlog/context.go]: Yes, even worse, the function was ignoring the skip parameter....

**Pull request merged**